lib/pr_checker.js

```diff
@@ -269,30 +269,39 @@ class PRChecker {
     } = this;
 
     if (reviews.length < 1) {
-      return true;
+      return false;
     }
 
-    const commitIndex = commits.length - 1;
     const reviewIndex = reviews.length - 1;
-    const lastCommit = commits[commitIndex].commit;
-    const lastReview = reviews[reviewIndex];
+    const reviewDate = reviews[reviewIndex].publishedAt;
 
-    const commitDate = lastCommit.committedDate;
-    const reviewDate = lastReview.publishedAt;
+    let afterCommits = [];
+    commits.forEach((commit) => {
+      commit = commit.commit;
+      if (commit.committedDate > reviewDate) {
+        afterCommits.push(commit);
+      }
+    });
 
-    let status = true;
-    if (commitDate > reviewDate) {
-      logger.warn('Changes were pushed since the last review:');
-      commits.forEach((commit) => {
-        commit = commit.commit;
-        if (commit.committedDate > reviewDate) {
-          status = false;
-          logger.warn(`- ${commit.message.split('\n')[0]}`);
-        }
+    let totalCommits = afterCommits.length;
+    if (totalCommits > 0) {
+      let warnMsg =
+        `${totalCommits} commits were pushed since the last review:`;
+
+      logger.warn(warnMsg.replace('1 commits were', 'Single commit was'));
+      if (totalCommits > 3) {
+        logger.warn('last 3 commits...');
+      }
+
+      afterCommits.splice(0, totalCommits - 3);
+      afterCommits.forEach(commit => {
```

Comment:
Or just `afterCommits.slice(totalCommits - 3).forEach(...)`?

Reply:
@joyeecheung this does not work because it would return the removed items rather than what we need the last three commits.

Reply:
`splice` would return the removed items while `slice` will return the sliced items...no?

Reply:
@cPhost `slice`ing an array with length < 3 would result in an empty array, if you are `forEach`ing the array I think it's gonna be fine.

Reply:
@joyeecheung the test are failing. I will give it a try again.

Reply:
@cPhost I think I know why it's failing: instead of `afterCommits.slice(totalCommits - 3)`, do `afterCommits.slice(-3)`, so you don't end up with `-2` or `-1`